### PR TITLE
samples: nrf: battery: extend to support measuring Vdd directly

### PR DESCRIPTION
**Pull request by pabigot**
When the system is powered directly a measurement of Vdd provides the battery voltage.  This requires a different ADC configuration and level curve.

Originally provided as an alternative to support #22414, now updated to replace that PR.

### CODE REVIEW

**Review by carlescufi — COMMENTED** [review]
Seems reasonable to me, at some point we really need to introduce an API for battery measurement though. I will let @anangl review and approve.

File: samples/boards/nrf/battery/src/battery.c [hunk 1]
```diff
@@ -46,17 +46,25 @@ struct gpio_channel_config {
 struct divider_config {
 	const struct io_channel_config io_channel;
```
Comment (anangl) [inline samples/boards/nrf/battery/src/battery.c]: These `const` modifiers here are actually redundant, as the whole instance of the structure is then created as `const`.

**Review by anangl — APPROVED** [review]
LGTM

**Comment by pabigot** [comment]
> Seems reasonable to me, at some point we really need to introduce an API for battery measurement though.

Agreed, but it'll be hard to do that until we can configure an ADC without referencing platform-specific information.

**Review by carlescufi — APPROVED** [review]

**Comment by pabigot** [comment]
Rebased in hopes this will be merged soon.

**Pull request merged**